Runar/Runar/Services/Notifications/LocalNotificationManager.swift

```diff
@@ -0,0 +1,117 @@
+
+        guard let weekday = dateComponents.weekday,
+              let hour = dateComponents.hour,
+              let week = dateComponents.weekOfMonth else { return }
+
+        if weekday == weekdaySetPoint && hour < hourSetPoint {
+            var identifier: String = .notificationGeneralIdentifier
+            identifier += "_Week_\(week)"
+            notificationCenter.removePendingNotificationRequests(withIdentifiers: [identifier])
+        }
+    }
+
+    /// Request permission from the user to receive notifications
+    private func requestAuthorization() {
+        notificationCenter.requestAuthorization(options: [.alert, .sound, .badge]) { [weak self] granted, error in
+            guard let self = self else { return }
+
+            if granted {
+                self.addRequest()
+            } else {
+                print("Local Notification: User has declined notifications")
+            }
+        }
+    }
+
+    /// The function of generating sending notifications
+    private func addRequest() {
+        // Notification Content
+        let content = UNMutableNotificationContent()
+        content.title = .notificationGeneralContentTitle
+        content.body = .notificationGeneralContentBody
+        content.sound = UNNotificationSound.default
+        content.badge = 1
+
+        // Configureting the recurring date
+        var dateComponents = DateComponents()
+        dateComponents.calendar = Calendar.current
+        dateComponents.weekday = TimeOfSendingLocalNotifications.weekday.rawValue
+        dateComponents.hour = TimeOfSendingLocalNotifications.hour.rawValue
+
+        for week in 1...5 {
+            var identifier: String = .notificationGeneralIdentifier
+            identifier += "_Week_\(week)"
+            dateComponents.weekOfMonth = week
+
+            // Creating the trigger as a repeating event
+            let trigger = UNCalendarNotificationTrigger(dateMatching: dateComponents, repeats: true)
+
+            // Creating the notification request
+            let request = UNNotificationRequest(identifier: identifier,
+                                                content: content,
+                                                trigger: trigger)
+
+            notificationCenter.add(request) { error in
+                if let error = error {
+                    print("Local Notification: Error \(error.localizedDescription)")
+                }
+            }
+        }
+
+        checkTheDayAndTime()
+    }
+
+    func addNotification() {
+        notificationCenter.getNotificationSettings { [weak self] settings in
+            guard let self = self else { return }
+
+            if settings.authorizationStatus == .authorized {
+                self.addRequest()
+            } else {
+                self.requestAuthorization()
+            }
+        }
+    }
+
+    func removeAllNotifications() {
+        notificationCenter.removeAllDeliveredNotifications()
```

Comment:
Why are we want to remove already delivered notifications?

Reply:
Just cleaning all notifications. If it is better to save the delivered notifications, then we can do so.